packages/context/src/context.ts

```diff
@@ -203,6 +203,27 @@ export class Context {
     }
     return json;
   }
+
+  /**
+   * Whether or not this context instance has a parent context instance.
+   *
+   * @returns {boolean}
+   * @memberof Context
+   */
+  hasParent(): boolean {
+    return !!this._parent;
+  }
+
+  /**
+   * Directly set the parent context of this context.
+   *
+   * @param {Context} ctx
+   * @memberof Context
+   */
+  setParent(ctx: Context) {
+    // FIXME(kev): This is definitely open to circular linking (bad!)
+    this._parent = ctx;
+  }
```

Comment:
If we have to make `_parent` writable, I would rather just change `parent` to be public. 

IMO, switching parent for a context is not desirable.

Reply:
It's not necessarily meant for _swapping_ context parents, but rather taking an orphan and giving it a parent. The use-case here is unique because we need to create the context at a time _before_ we have access to the higher-level context instance. This allows us to lazy-assign the parent to our subcontext.

Reply:
Also, I'm adding some circular link checking to this which should make it safer than exposing the property directly.